sdk/metric/metricdata/metricdatatest/assertion.go

```diff
@@ -44,38 +44,66 @@ type Datatypes interface {
 	// Aggregation and Value type from metricdata are not included here.
 }
 
+type config struct {
+	ignoreTimestamp bool
+}
+
+// Option allows for fine grain control over how AssertEqual operates.
+type Option interface {
+	apply(cfg config) config
+}
+
+type fnOption func(cfg config) config
+
+func (fn fnOption) apply(cfg config) config {
+	return fn(cfg)
+}
+
+// IgnoreTimestamp disables checking if timestamps are different.
+func IgnoreTimestamp() Option {
+	return fnOption(func(cfg config) config {
+		cfg.ignoreTimestamp = true
+		return cfg
+	})
+}
+
 // AssertEqual asserts that the two concrete data-types from the metricdata
 // package are equal.
-func AssertEqual[T Datatypes](t *testing.T, expected, actual T) bool {
+func AssertEqual[T Datatypes](t *testing.T, expected, actual T, opts ...Option) bool {
 	t.Helper()
 
+	cfg := config{}
+	for _, opt := range opts {
+		cfg = opt.apply(cfg)
+	}
+
 	// Generic types cannot be type asserted. Use an interface instead.
 	aIface := interface{}(actual)
 
 	var r []string
 	switch e := interface{}(expected).(type) {
 	case metricdata.DataPoint[int64]:
-		r = equalDataPoints(e, aIface.(metricdata.DataPoint[int64]))
+		r = equalDataPoints(e, aIface.(metricdata.DataPoint[int64]), cfg.ignoreTimestamp)
```

Comment:
nit: should we just pass cfg to make our lives easier in the future?

Reply:
I debated if we should go the other way and not have an `Option` and `config`.  I think it should happen if we have another option, but there just ends up with more boilerplate, especially in the tests, passing in the whole config.

If there is a push for it I can change it.

Reply:
I'm totally on-board with having Option and config.  I was just suggesting changing the function signature from
`func equalDataPoints(... ignoreTimestamp bool) []string` to `func equalDataPoints(... cfg config) []string`, or alternatively `func (c *config) equalDataPoints(...) []string`.